lms/views/oauth.py

```diff
@@ -15,11 +18,10 @@ def build_canvas_token_url(lms_url):
 
 
 @view_config(route_name="canvas_oauth_callback", request_method="GET")
+@parser.use_kwargs(CANVAS_OAUTH_CALLBACK_ARGS)
 # pylint: disable=too-many-locals
-def canvas_oauth_callback(request):
+def canvas_oauth_callback(request, state):
```

Comment:
All you have to do to add validation to a view is add a decorator to it like this. The parsed and valid args are passed to the view as separate arguments (if there are lots of them you can use `**kwargs` here). If validation fails the view isn't called.

By default webargs will look for arguments in the request's query params, form data, and JSON data, in that order. You can pass arguments to the view decorator to restrict where arguments can come from, e.g. query params only, or only query params or JSON body. You can also add new locations, headers, cookies and files are supported but disabled by default. I think you can write custom fetchers too.

For this view I decided the default of query, form and JSON is fine.